### PR TITLE
Migrate to Vite

### PR DESCRIPTION
**Pull request by ThibautBremand**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -3,22 +3,17 @@
   "version": "1.0.0",
   "description": "Monorepo related to the GemWallet extension",
   "private": true,
+  "repository": "https://github.com/GemWallet/gemwallet-extension.git",
```
Comment (FlorianBouron) [inline package.json:6]: Why isn't it an object anymore?

File: package.json [hunk 2]
```diff
@@ -3,22 +3,17 @@
...
   "scripts": {
-    "build": "NODE_OPTIONS=--openssl-legacy-provider yarn workspace @gemwallet/extension build",
+    "prepare": "husky install",
+    "build": "yarn workspace @gemwallet/extension build",
     "build:api:cdn": "yarn workspace @gemwallet/api build:cdn",
     "build:api:npm": "yarn workspace @gemwallet/api build:npm",
-    "cypress": "yarn workspace @gemwallet/extension cypress:e2e",
-    "lint": "eslint --cache --quiet --fix",
```
Comment (FlorianBouron) [inline package.json:11]: Why was the linting command removed?

File: package.json [hunk 3]
```diff
@@ -3,22 +3,17 @@
...
-    "prepare": "husky install",
-    "prettier": "prettier --write **/*.{js,ts,tsx}",
-    "release:api:npm": "cd packages/api/dist && npm publish --access=public",
```
Comment (FlorianBouron) [inline package.json:14]: It seems that you didn't add this script back

File: packages/extension/.gitignore [hunk 4]
```diff
@@ -1,26 +1,24 @@
...
-
-# dependencies
-/node_modules
-/.pnp
-.pnp.js
-
-# testing
-/coverage
-
-# production
```
Comment (FlorianBouron) [inline packages/extension/.gitignore:11]: The production folder should be added back, if you create a build it will want to be added to the repository

File: packages/extension/.gitignore [hunk 5]
```diff
@@ -1,26 +1,24 @@
...
-/build
-
-# misc
-.DS_Store
-.env.local
```
Comment (FlorianBouron) [inline packages/extension/.gitignore:16]: All the .env files need to be added back as we use them and we don't want them to be commuted with the secrets

File: packages/extension/.gitignore [hunk 6]
```diff
@@ -1,26 +1,24 @@
-# See https://help.github.com/articles/ignoring-files/ for more about ignoring files.
```
Comment (FlorianBouron) [inline packages/extension/.gitignore:1]: Minor: I would keep this comment

File: packages/extension/README.md [hunk 7]
```diff
@@ -1,58 +1,30 @@
-# GemWallet Extension
```
Comment (FlorianBouron) [inline packages/extension/README.md:1]: I would keep this for SEO purpose

File: packages/extension/src/chromeServices/background/index.ts [hunk 8]
```diff
@@ -289,7 +292,9 @@ chrome.runtime.onMessage.addListener(
           requestMessage: message.type,
           sender
         });
-      } catch (e) {}
+      } catch (e) {
+        console.log(e);
```
Comment (FlorianBouron) [inline packages/extension/src/chromeServices/background/index.ts:296]: Maybe console.error(e)
Same for all the console logs which were added

File: packages/extension/src/components/organisms/TransactionListing/format.util.ts [hunk 9]
```diff
@@ -1,4 +1,4 @@
-import { unix } from 'moment';
```
Comment (FlorianBouron) [inline packages/extension/src/components/organisms/TransactionListing/format.util.ts:1]: Why did you remove moment.js?

Comment (ThibautBremand) [reply packages/extension/src/components/organisms/TransactionListing/format.util.ts:1]: Because it's a deprecated library which is not supported by vite, dayjs is the new go-to

File: package.json (hunk 1 above)
Comment (ThibautBremand) [reply package.json:6]: It was generated like this when I set up the repo

File: package.json [hunk 10]
```diff
@@ -43,10 +38,5 @@
     "**/*.{json,md,html}": [
       "prettier --write"
     ]
-  },
-  "engines": {
```
[the remark this answers is not in the document]
Comment (ThibautBremand) [reply package.json:47]: Because they were not put when I set up the vite project

File: packages/extension/.gitignore (hunk 4 above)
Comment (ThibautBremand) [reply packages/extension/.gitignore:11]: With this config, the build is generated into the `dist` directory

File: package.json (hunk 10 above)
[the remark this answers is not in the document]
Comment (ThibautBremand) [reply package.json:47]: Done

File: packages/extension/package.json [hunk 11]
```diff
@@ -3,78 +3,70 @@
...
-    "xrpl-accountlib": "^3.2.6",
-    "xrpl-secret-numbers": "^0.3.4"
+    "stream-browserify": "^3.0.0",
+    "web-vitals": "^3.5.1",
+    "xrpl": "^3.0.0",
+    "xrpl-accountlib": "^3.2.9"
   },
   "devDependencies": {
     "@gemwallet/constants": "*",
     "@gemwallet/eslint-config": "*",
```
Comment (FlorianBouron) [inline packages/extension/package.json:45]: You are pulling our eslint configuration, though we can see in the dev dependencies that you are using the dependencies from vite.js as you require another version of eslint.
We have two solutions here:

- We use the configuration from vite.js for the linter on the extension and the package @gemwallet/eslint-config for the API
- We update our configuration within @gemwallet/eslint-config to match vite and override it, and we make use of it within the extension.

We can also do this in another MR to don't add more change as this is not blocker.

What do you think?

**Review by FlorianBouron — APPROVED** [review]
